pkg/controller/dataimportcron-controller.go

```diff
@@ -0,0 +1,594 @@
+		err := r.cleanup(ctx, dataImportCron)
+		return reconcile.Result{}, err
+	}
+	if dataImportCron.Annotations[AnnCronInitialized] != "true" {
+		return r.initCron(ctx, dataImportCron)
+	}
+	return r.update(ctx, dataImportCron)
+}
+
+func (r *DataImportCronReconciler) initCron(ctx context.Context, dataImportCron *cdiv1.DataImportCron) (reconcile.Result, error) {
+	res := reconcile.Result{}
+	if dataImportCron.Annotations == nil {
+		dataImportCron.Annotations = make(map[string]string)
+	}
+
+	regSource, err := getCronRegistrySource(dataImportCron)
+	if err != nil {
+		return res, err
+	}
+
+	if regSource.ImageStream != nil {
+		if err := r.updateImageStreamDesiredDigest(ctx, dataImportCron); err != nil {
+			return res, err
+		}
+		res, err = r.setNextCronTime(dataImportCron)
+		if err != nil {
+			return res, err
+		}
+	} else if regSource.URL != nil {
+		cronJob, err := r.newCronJob(dataImportCron)
+		if err != nil {
+			return res, err
+		}
+		if err := r.client.Create(ctx, cronJob); err != nil {
+			r.log.Error(err, "Unable to create CronJob")
+			return res, err
+		}
+		AddFinalizer(dataImportCron, dataImportCronFinalizer)
+	}
+
+	dataImportCron.Annotations[AnnCronInitialized] = "true"
+	if err := r.client.Update(ctx, dataImportCron); err != nil {
+		r.log.Error(err, "Unable to update DataImportCron", "Name", dataImportCron.Name)
+		return res, err
+	}
+	return res, nil
+}
+
+func (r *DataImportCronReconciler) getImageStream(ctx context.Context, imageStreamName, imageStreamNamespace string) (*imagev1.ImageStream, error) {
+	if imageStreamName == "" || imageStreamNamespace == "" {
+		return nil, errors.Errorf("Missing imagestream name or namespace")
+	}
+	imageStream := &imagev1.ImageStream{}
+	imageStreamNamespacedName := types.NamespacedName{
+		Namespace: imageStreamNamespace,
+		Name:      imageStreamName,
+	}
+	if err := r.client.Get(ctx, imageStreamNamespacedName, imageStream); err != nil {
+		return nil, err
+	}
+	return imageStream, nil
+}
+
+func getImageStreamDigest(imageStream *imagev1.ImageStream) (string, error) {
+	if imageStream == nil {
+		return "", errors.Errorf("No imagestream")
+	}
+	tags := imageStream.Status.Tags
+	if len(tags) == 0 || len(tags[0].Items) == 0 {
+		return "", errors.Errorf("No imagestream tag items %s", imageStream.Name)
+	}
+	return tags[0].Items[0].Image, nil
+}
+
+func (r *DataImportCronReconciler) pollImageStreamDigest(ctx context.Context, dataImportCron *cdiv1.DataImportCron) (reconcile.Result, error) {
+	nextTime, err := time.Parse(time.RFC3339, dataImportCron.Annotations[AnnNextCronTime])
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	if nextTime.Before(time.Now()) {
+		desiredDigest := dataImportCron.Annotations[AnnSourceDesiredDigest]
+		if desiredDigest != "" && desiredDigest != dataImportCron.Status.CurrentImportDigest {
+			dataImportCron.Annotations[AnnSourceUpdatePending] = "true"
+		}
+		return r.setNextCronTime(dataImportCron)
+	}
+	return reconcile.Result{}, nil
```

Comment:
Not sure it's a must, as the previously returned `requeueAfter` should cause a reclaim as scheduled (so `nextTime.Before(time.Now())` is true). Adding it just to be on the safe side.

Reply:
That's not how `requeueAfter` works.  If a resource is updated before the `requeueAfter` time, it will be reconciled again immediately and not again (unless requeueAfter is set again)